cmd/lambda/accounts/delete.go

```diff
@@ -1,113 +1,28 @@
 package main
 
 import (
-	"encoding/json"
-	"fmt"
-	"log"
 	"net/http"
 
-	"github.com/Optum/dce/pkg/rolemanager"
-	"github.com/aws/aws-sdk-go/service/iam"
-
-	"github.com/Optum/dce/pkg/api/response"
-	"github.com/Optum/dce/pkg/common"
-	"github.com/Optum/dce/pkg/db"
+	"github.com/Optum/dce/pkg/api"
 	"github.com/gorilla/mux"
 )
 
 // DeleteAccount - Deletes the account
 func DeleteAccount(w http.ResponseWriter, r *http.Request) {
 
 	accountID := mux.Vars(r)["accountId"]
-	deletedAccount, err := Dao.DeleteAccount(accountID)
-
-	// Handle DB errors
-	if err != nil {
-		switch err.(type) {
-		case *db.AccountNotFoundError:
-			response.WriteNotFoundError(w)
-			return
-		case *db.AccountLeasedError:
-			response.WriteAPIErrorResponse(
-				w,
-				http.StatusConflict,
-				"Conflict",
-				err.Error(),
-			)
-			return
-		default:
-			response.WriteServerErrorWithResponse(w, "Internal Server Error")
-			return
-		}
-	}
-
-	// Delete the IAM Principal Role for the account
-	destroyIAMPrincipal(deletedAccount)
-
-	// Publish SNS "account-deleted" message
-	sendSNS(deletedAccount)
-
-	// Push the account to the Reset Queue, so it gets cleaned up
-	sendToResetQueue(deletedAccount)
-
-	// json.NewEncoder(w).Encode(response.CreateAPIGatewayResponse(http.StatusNoContent, ""))
-	response.WriteAPIResponse(w, http.StatusNoContent, "")
-}
-
-// sendSNS sends notification to SNS that the delete has occurred.
-func sendSNS(account *db.Account) {
-	serializedAccount := response.AccountResponse(*account)
-	serializedMessage, err := common.PrepareSNSMessageJSON(serializedAccount)
 
+	acct, err := Services.Config.AccountService().Get(accountID)
```

Comment:
Might be worth calling out in the `Get()` method what kind of errors it returns
(I just had to dig in a bit to figure out that we're actually generating different HTTP error types at the data layer)